src/components/coupon/CouponPlanDescriptionTabs.tsx

```diff
@@ -126,7 +128,13 @@ const CouponPlanDescriptionTabs: React.FC<{
                 )}`}
               </Typography.Text>
               <AdminModal
-                renderTrigger={({ setVisible }) => <DeleteOutlined className="ml-4" onClick={() => setVisible(true)} />}
+                renderTrigger={({ setVisible }) =>
+                  Boolean(permissions.COUPON_PLAN_ADMIN) || Boolean(permissions.COUPON_PLAN_ADMIN_DELETE) ? (
+                    <DeleteOutlined className="ml-4" onClick={() => setVisible(true)} />
+                  ) : (
+                    <></>
```

Comment:
While `<></>` and `null` are generally implemented similarly, we can discuss which one we might prefer

Reply:
"<></>" offers higher readability for the code.